### PR TITLE
Card game peter pan tests

### PR DESCRIPTION
**Pull request by matsarat**

### CODE REVIEW

File: src/test/java/com/example/java/maven/cardGamePeterPan/GameTest.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        then(messagePrinter)
+                .should(times(1))
+                .printPlayer(playerList.get(1));
+        then(messagePrinter)
+                .should(times(1))
+                .printPlayer(playerList.get(2));
+    }
+
+    @Test
+    void shouldDiscardCards() {
```
Comment (Czeffik) [inline src/test/java/com/example/java/maven/cardGamePeterPan/GameTest.java:66]: missing case - what if cards not matching?

Comment (Czeffik) [inline src/test/java/com/example/java/maven/cardGamePeterPan/GameTest.java:66]: missing cases - what if `IndexOutOfBoundsException` or `IllegalArgumentException` thrown?

File: src/test/java/com/example/java/maven/cardGamePeterPan/GameTest.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                .printMessage(ASK_FOR_NEXT_CARD_TO_DISCARD);
+
+
+        assertThat(player.getHandSize()).isEqualTo(1);
+        assertThat(player.getHand().get(0)).isEqualTo(card3);
+
+    }
+
+    @Test
+    void shouldTakeChosenCardFromPreviousPlayerHand() {
```
Comment (Czeffik) [inline src/test/java/com/example/java/maven/cardGamePeterPan/GameTest.java:116]: missing case - what if `IndexOutOfBoundsException` is thrown?

File: src/test/java/com/example/java/maven/cardGamePeterPan/GameTest.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+import org.mockito.Mockito;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.Mockito.times;
+
+public class GameTest {
```
Comment (Czeffik) [inline src/test/java/com/example/java/maven/cardGamePeterPan/GameTest.java:14]: missing tests for `play` method? :D

File: src/test/java/com/example/java/maven/cardGamePeterPan/DeckTest.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    @BeforeAll
+    public static void createDeck() {
+        deck = new Deck();
+    }
+
+    @Test
+    void deckShouldIncludeRequiredNumberOfCards() {
+
+//        then
+        assertThat(deck.getCards().size()).isEqualTo(25); // because we have 24 card in small deck (from 9 to Ace) + Witch
```
Comment (Czeffik) [inline src/test/java/com/example/java/maven/cardGamePeterPan/DeckTest.java:24]: and should contains `WITCH`  and cards should be unique

File: src/test/java/com/example/java/maven/cardGamePeterPan/PlayerTest.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    void shouldAddCardToPlayersHand() {
+
+//        given
+        Card card = Mockito.mock(Card.class);
+
+//        when
+        player.addCardToHand(card);
+
+//        then
+        assertThat(player.getHandSize()).isEqualTo(1);
```
Comment (Czeffik) [inline src/test/java/com/example/java/maven/cardGamePeterPan/PlayerTest.java:29]: and contains `card`

File: src/test/java/com/example/java/maven/cardGamePeterPan/PlayerTest.java [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+    }
+
+    @Test
+    void shouldTakeCardFromAnotherPlayersHand() {
```
Comment (Czeffik) [inline src/test/java/com/example/java/maven/cardGamePeterPan/PlayerTest.java:34]: NICE!

File: src/test/java/com/example/java/maven/cardGamePeterPan/PlayerTest.java [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+//        given
+        Card card1 = new Card(Card.Suit.CLUB, Card.Rank.ACE);
+        Card card2 = new Card(Card.Suit.SPADE, Card.Rank.JACK);
+        Card card3 = new Card(Card.Suit.CLUB, Card.Rank.TEN);
+
+//        when
+        player.addCardToHand(card1);
+        player.addCardToHand(card2);
+        player.addCardToHand(card3);
```
Comment (Czeffik) [inline src/test/java/com/example/java/maven/cardGamePeterPan/PlayerTest.java:61]: MINOR:
adding cards is a part of test preparation - `given` section. In this test you are testing `isDiscardingCardsPossible()` method

**Review by Czeffik — COMMENTED** [review]
GOOD JOB!

File: src/test/java/com/example/java/maven/cardGamePeterPan/PlayerTest.java (hunk 7 above)
Comment (matsarat) [reply src/test/java/com/example/java/maven/cardGamePeterPan/PlayerTest.java:61]: Done

File: src/test/java/com/example/java/maven/cardGamePeterPan/PlayerTest.java (hunk 5 above)
Comment (matsarat) [reply src/test/java/com/example/java/maven/cardGamePeterPan/PlayerTest.java:29]: Done

File: src/test/java/com/example/java/maven/cardGamePeterPan/DeckTest.java (hunk 4 above)
Comment (matsarat) [reply src/test/java/com/example/java/maven/cardGamePeterPan/DeckTest.java:24]: Done

File: src/test/java/com/example/java/maven/cardGamePeterPan/GameTest.java (hunk 2 above)
Comment (matsarat) [reply src/test/java/com/example/java/maven/cardGamePeterPan/GameTest.java:116]: Done

**Pull request merged**